first_ruby/app/assets/stylesheets/application.css

```diff
@@ -0,0 +1,15 @@
+/*
+ * This is a manifest file that'll be compiled into application.css, which will include all the files
+ * listed below.
+ *
+ * Any CSS and SCSS file within this directory, lib/assets/stylesheets, or any plugin's
+ * vendor/assets/stylesheets directory can be referenced here using a relative path.
+ *
+ * You're free to add application-wide styles to this file and they'll appear at the bottom of the
+ * compiled file so the styles you add here take precedence over styles defined in any other CSS/SCSS
+ * files in this directory. Styles in this file should be added after the last require_* statement.
+ * It is generally better to create a new file per style scope.
+ *
+ *= require_tree .
+ *= require_self
+ */
```

Comment:
How does this file load all css & scss files within this directory as stated in line 5?

Reply:
This is all done by Asset Pipeline which is implemented by sprocket-rails gem which is enabled by default.

Reply:
You are basically telling me here is magic.
I am asking you how does the magic work.
So, if I ask you to add a new css file, will it get loaded?
If the css file is from a gem, it will automatically get loaded?

Reply:
**Asset Pipeline** provides feature such as concatenate, minify or compression of JavaScript and CSS assets, this is to
improve the performance by reducing number of requests render to the web page. The feature of asset pipeline are:-
1) **Concatenate** does this by concatenate all JavaScript file into one master `.js` file and CSS into one master `.css` file. 
2) The **minification** or **compression** feature is by removing whitespace and comments in CSS file, but slightly more complex for JavaScript.
3) It also allow **precompilation** down to actual asset which include Sass for CSS, CoffeeScript for Javascript and ERB for both by default.
4) Also allow caching of files in the asset, which enhance performance.

> So, if I ask you to add a new css file, will it get loaded?

It must be added inside the `assets/stylesheet` to get loaded.

> If the css file is from a gem, it will automatically get loaded?

Since it's from a gem, it is usually inside the `vendor` but it must be inside `vendor/asset`, since asset pipeline will look for inside `app/asset` , `lib/asset` and `vendor/asset`.

Reply:
1-4: ok

> Since it's from a gem, it is usually inside the vendor but it must be inside vendor/asset, since asset pipeline will look for inside app/asset , lib/asset and vendor/asset.
What if you need to use the library in your scss / css?

Reply:
> What if you need to use the library in your scss / css?

In this case, in the current scss/css, use the `@import` keyword, for example `@import 'directory/example.scss'` inside the scss/css file.

Reply:
you can only use @import if its scss. if its css, you can't

Reply:
For `.css` file, `@import` can be use too.